crates/matrix-sdk-crypto/src/gossiping/machine.rs

```diff
@@ -1039,11 +1037,11 @@ mod tests {
         ReadOnlyAccount::new(alice_id(), alice2_device_id())
     }
 
-    fn bob_machine() -> GossipMachine {
-        let user_id = Arc::from(bob_id());
-        let account = ReadOnlyAccount::new(&user_id, alice_device_id());
+    fn bob_machine(user_id: &UserId) -> GossipMachine {
+        let user_id = Arc::from(user_id);
+        let account = ReadOnlyAccount::new(&user_id, bob_device_id());
```

Comment:
I find this pretty confusing, maybe because I am not familiar with the API being tested. When I see `ReadOnlyAccount::new(user_id, device_id)`, it looks to me as if the device ID would naturally be of a device for that user, but here they seem to be mostly unrelated?

Reply:
The only requirement here is that the device id is unique, we don't use more than two device ids in the whole test module so that requirement is met.

Perhaps it would be better if we generate a random device id?

Reply:
If that makes the test work the same way, it seems preferable to me.

Reply:
Not complicated at all: https://github.com/matrix-org/matrix-rust-sdk/pull/901/commits/1d27d72c8f1f9d7e040acfe9e5b6af36ad6a2e62.